internal/common/middleware/auth.go

```diff
@@ -0,0 +1,73 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"kg/procurement/internal/token"
+	"net/http"
+	"strings"
+)
+
+const (
+	AuthorizationHeader = "Authorization"
+	BearerType          = "Bearer"
+	ErrorAuthHeader     = "authorization header not provided"
+	ErrorAuthInvalid    = "authorization header not valid"
+	ErrorAuthType       = "authorization type not valid"
+)
+
+type AuthMiddleware struct {
+	tokenManager token.TokenManager
+}
+
+func NewAuthMiddleware(manager token.TokenManager) *AuthMiddleware {
+	return &AuthMiddleware{
+		tokenManager: manager,
+	}
+}
+
+func (m *AuthMiddleware) MustAuthenticated() gin.HandlerFunc {
```

Comment:
Do you think we could rename this to `MustBeAuthenticated` for grammatical reasons

Reply:
Alas! My english is not that good!